# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1293466386 18000
#      Mon Dec 27 11:13:06 2010 -0500
# Node ID 74e285bb61c958dc153bc12e9aabd2e45b1d424f
# Parent  e7ed20f87e8277839924d3945bb64aa5c6945dc2
text.m: Add demo for text rotation and alignment.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-12-27  Ben Abbott  <bpabbott@mac.com>
+
+	* plot/text.m: Add demo for text rotation and alignment.
+
 2010-12-26  Michael Godfrey <godfrey@isl.stanford.edu>
 
 	* specfun/legendre.m: Add @tex blocks to docstring.
 
 2010-12-24  Rik  <octave@nomad.inbox5.com>
 
 	* testfun/test.m: Use 'Command' for deftype in docstring.
 
diff --git a/scripts/plot/text.m b/scripts/plot/text.m
--- a/scripts/plot/text.m
+++ b/scripts/plot/text.m
@@ -98,8 +98,33 @@ function h = text (varargin)
       h = tmp;
     endif
 
   else
     print_usage ();
   endif
 
 endfunction
+
+%!demo
+%! clf
+%! ha = {"left", "center", "right"};
+%! va = {"bottom", "middle", "top"};
+%! for t = 0:30:359;
+%!   for nh = 1:numel(ha)
+%!     x = [0.25 0.5 0.75](nh);
+%!     for nv = 1:numel(va)
+%!       y = [0.25 0.5 0.75](nv);
+%!       text (x, y, "Hello World",
+%!                   "rotation", t,
+%!                   "horizontalalignment", ha{nh},
+%!                   "verticalalignment", va{nv})
+%!     endfor
+%!   endfor
+%! endfor
+%! set (gca, "xtick", [0.25, 0.5, 0.75],
+%!           "xticklabel", ha,
+%!           "ytick", [0.25, 0.5, 0.75],
+%!           "yticklabel", va)
+%! axis ([0 1 0 1])
+%! xlabel ("horizontal alignment")
+%! ylabel ("vertical alignment")
+%! title ("text alignment and rotation (0:30:360 degrees)")
